BootloaderCommonPkg/Library/ShellLib/CmdBoot.c

```diff
@@ -403,13 +403,15 @@ PrintBootOption (
                  BootOption->Image[0].LbaImage.LbaAddr \
                  );
     }
-
     for (ExtraIndex = 1; ExtraIndex < LoadImageTypeMax; ExtraIndex++) {
-      if (BootOption->Image[ExtraIndex].LbaImage.Valid == 1) {
-        ShellPrint (L" [%x|0x%x]", BootOption->Image[ExtraIndex].LbaImage.SwPart, \
-          BootOption->Image[ExtraIndex].LbaImage.LbaAddr);
-      } else if (BootOption->Image[ExtraIndex].FileName[0] != '\0') {
-        ShellPrint (L" [%a]", BootOption->Image[ExtraIndex].FileName);
+      if ((((BootOption->BootFlags & BOOT_FLAGS_PREOS) == BOOT_FLAGS_PREOS) && (ExtraIndex == LoadImageTypePreOs))

```

Comment:
PREOS used Image[1], it is not part of extra Images. It is confused to have this check under a extra image "for" loop.

Reply:
before this patch the for loop itself it still will print PREOS filename right?
should we modify the loop to fit for PREOS and extra image together?

Reply:
no. before this ExtraIndex for loop, it only print the normal OS using Image[0].
you could add Image[1] print before this for loop when PreOS flag is set.

Reply:
Updated patch